docs/docsite/rst/lookup_guide.rst

```diff
@@ -0,0 +1,159 @@
+
+Generic read (not ``kv`` specific) is still important functionality, so we have the ``community.hashi_vault.vault_read`` :ref:`module <ansible_collections.community.hashi_vault.vault_read_module>` and :ref:`lookup plugin <ansible_collections.community.hashi_vault.vault_read_lookup>` to provide that without trying to infer whether the response is from a specific backend.
+
+Since reading from ``kv`` store is by far the most common use case, we will also be introducing content for that specifically, which will accept familiar paths and provide parameters for ``kv``-specific functionality like versioning. That content is coming soon.
+
+The dictionary dereferencing via ``:keyname`` syntax *will not be supported* in other content. That will be achieved in Jinja via:
+
+* dot syntax ``.keyname``
+* lookup syntax ``['keyname']``
+* specialized filters in some circumstances, such as the ``vault_login_token`` :ref:`filter <ansible_collections.community.hashi_vault.docsite.filter_guide.vault_login_token>`.
+
+Parameters via term string *will not be supported* in other lookups. Its use is discouraged by core developers, and steps have already been taken in core to remove the functionality where it still exists, however it will remain in the ``hashi_vault`` plugin for backwards compatibility and because it is likely to still be in use in a lot of places.
+
+The future of the ``hashi_vault`` lookup
+----------------------------------------
+
+There are no plans currently to deprecate or remove the ``hashi_vault`` plugin. It is likely that it will stay indefinitely, for backwards compatibility and because so much functionality has been moved to shared code that very little maintenance is required to keep it. This decision may be revisited if circumstances change.
+
+That being said, we will encourage the use of newer content that has functionality with a tighter scope and is expected to receive updates and enchancements as appropriate.
+
+New features and functionality are unlikely to be added or accepted in the ``hashi_vault`` lookup, except for the ones that come for "free", like new auth methods (these require no code changes to the plugin itself).
+
+Lookups and writes
+==================
+
+Lookups in Ansible kind of imply a read-only/non-destructive operation. Being run in templating, the expected result is that they *return* a value, and check mode has no effect on lookups. But there is no actual restriction on what a lookup can do, and some lookups do perform writes or otherwise change some state. For example, the ``password`` :ref:`lookup <ansible_collections.ansible.builtin.password_lookup>` writes a generated password to a file, to act as a sort of cache, and the ``pipe`` :ref:`lookup <ansible_collections.ansible.builtin.pipe_lookup>` runs an arbitrary shell command so it could easy write or change state.
```

Comment:
Okay, I'm not sure I'm following this - but I think what you're saying is:
```suggestion

Most Ansible lookups perform read-only, non-destructive operations. They are run in templating, they generally *return*  values, and they do not generally run in check mode. However, some lookups do change state, sometimes by performing write operations. For example, the ``password`` :ref:`lookup <ansible_collections.ansible.builtin.password_lookup>` writes a generated password to a file, to act as a sort of cache, and the ``pipe`` :ref:`lookup <ansible_collections.ansible.builtin.pipe_lookup>` runs an arbitrary shell command so it could easy write or change state. The ``hashi_vault`` lookup also performs write operations.
```

Reply:
Yeah your rewrite is correct, with a few small details I can update:
* lookups _do_ run in check mode (as in, they perform whatever they were going to do, blissfully unaware that your intention was to run ansible in a way that doesn't make changes)
* this section isn't about the `hashi_vault` plugin, even though it along with the other lookups in the collection do perform writes of a sort, implicitly. This is just a brief overview/reminder(?) of the way lookups work and that they imply read _by convention_ but are not limited to that.

Ultimately this, and the next section (about writes in Vault generally), are the pre-requisites to understanding how to properly use lookups in a way that you are not surprised by the behavior, and can understand when things that don't look like writes can have an effect at scale (if you run tens of thousands of lookups that all seem like reads to you, you might be surprised when Vault runs into storage/replication issues due to a sudden flood of token creation for example).